contracts/bridge/contracts/LightClientBridge.sol

```diff
@@ -0,0 +1,429 @@
+     * @param beefyMMRLeafCount contains the merkle leaf count
+     * @param peaks contains the merkle maintain range peaks
+     * @param siblings contains the merkle maintain range siblings
+     */
+    function verifyBeefyMerkleLeaf(
+        bytes calldata beefyMMRLeaf,
+        uint256 beefyMMRLeafIndex,
+        uint256 beefyMMRLeafCount,
+        bytes32[] calldata peaks,
+        bytes32[] calldata siblings 
+    ) external view returns (bool) {
+        return
+            MMR.inclusionProof(
+                latestMMRRoot,
+                beefyMMRLeafCount,
+                beefyMMRLeafIndex,
+                beefyMMRLeaf,
+                peaks,
+                siblings
+            );
+    }
+
+    /**
+     * @notice Executed by the prover in order to begin the process of block
+     * acceptance by the light client
+     * @param commitmentHash contains the commitmentHash signed by the validator(s)
+     * @param validatorClaimsBitfield a bitfield containing a membership status of each
+     * validator who has claimed to have signed the commitmentHash
+     * @param validatorSignature the signature of one validator
+     * @param validatorPosition the position of the validator, index starting at 0
+     * @param validatorPublicKey the public key of the validator
+     * @param validatorPublicKeyMerkleProof proof required for validation of the public key in the validator merkle tree
+     */
+    function newSignatureCommitment(
+        bytes32 commitmentHash,
+        uint256[] memory validatorClaimsBitfield,
+        bytes memory validatorSignature,
+        uint256 validatorPosition,
+        address validatorPublicKey,
+        bytes32[] memory validatorPublicKeyMerkleProof
+    ) public payable whenNotPaused {
+        /**
+         * @dev Check if validatorPublicKeyMerkleProof is valid based on ValidatorRegistry merkle root
+         */
+        require(
+            checkValidatorInSet(
+                validatorPublicKey,
+                validatorPosition,
+                validatorPublicKeyMerkleProof
+            ),
+            "Error: Validator must be in validator set at correct position"
+        );
+
+        /**
+         * @dev Check if validatorSignature is correct, ie. check if it matches
+         * the signature of senderPublicKey on the commitmentHash
+         */
+        require(
+            ECDSA.recover(commitmentHash, validatorSignature) ==
+                validatorPublicKey,
+            "Error: Invalid Signature"
+        );
+
+        /**
+         * @dev Check that the bitfield actually contains enough claims to be succesful, ie, > 2/3
+         */
+        require(
+            validatorClaimsBitfield.countSetBits() >
+                (numOfValidators * THRESHOLD_NUMERATOR) /
+                    THRESHOLD_DENOMINATOR,
+            "Error: Bitfield not enough validators"
+        );
+
+        /**
+         * @todo Lock up the sender stake as collateral
+         */
+        // TODO
```

Comment:
TODO here, because if one of the validators(1/N) could initialize the `newSignatureCommitment` with wrong `commitmentHash` or `validatorClaimsBitfield` and block the later `completeSignatureCommitment` from finishing the protocol.

So we will need an collateral and slash on fault proof(or out of time)

Refer the optimistic scheme in this doc: https://hackmd.io/ohOt4jAPT8uu-soJXHUq0Q#Optimistic-scheme

Reply:
We can create an todo issue for this, and suggest finish it in another PR.

It is 1/N risk, we can still test/run in optimistic situation.

Reply:
https://github.com/darwinia-network/darwinia-bridge-sol/issues/42